src/app/form/form.component.html

```diff
@@ -0,0 +1,81 @@
+                    <p *ngIf="typeOfForm === 'coord'" class="mb-0">Coord</p>
+                  </span>
+                  <input id="city" name="city" class="form-control" type="text" formControlName="weatherCity" *ngIf="typeOfForm === 'city'"/>
+                  <input id="lat" name="lat" class="form-control" type="text" placeholder="Latitude"
+                      formControlName="weatherLat" *ngIf="typeOfForm === 'coord'"/>
+                  <input id="lon" name="lon" class="form-control" type="text" placeholder="Longitude"
+                      formControlName="weatherLon" *ngIf="typeOfForm === 'coord'"/>
+                </div>
+              </div>
+            </div>
+
+            <!-- Data Collected : Unit -->
+            <div class="row justify-content-center m-1">
+              <div class="col-4">
+                <div class="form-floating">
+                  <select id="units" class="form-select" name="units" id="floatingUnit"
+                          formControlName="weatherUnit"  bindQueryParam="unit" >
+                    <option value="metric"> metric </option>
+                    <option value="imperial"> imperial </option>
+                  </select>
+                  <label for="floatingUnit">Unit</label>
+                </div>
+              </div>
+
+              <!-- Data Collected : Langage (default = English) -->
+              <div class="col-4">
+                <div class="form-floating">
+                  <select id="langage" class="form-select" id="floatingLang" name="langage"
+                          formControlName="weatherLang">
+                    <option> en </option>
+                    <option> fr </option>
+                  </select>
+                  <label for="floatingLang">Lang</label>
+                </div>
+              </div>
+            </div>
+
+
+            <div class="row justify-content-center">
+              <div class="col-2 m-2">
+                <button type="submit" class="btn btn-primary bg-primary"  [disabled]="weatherForm!.invalid" >
+
+                  <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor"
+                       class="bi bi-search" viewBox="0 0 16 16">
+                    <path
+                      d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z" />
+                  </svg>
+                </button>
+              </div>
+            </div>
+          </form>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+<app-display *ngIf="!formResearch"
+  [typeOfForm]="typeOfForm"
+  [cityNameFromForm]="cityFormValue"
+  [latFromForm]="latFormValue"
+  [lonFromForm]="lonFormValue"
+  [unitFromForm]="unitFormValue"
+  [langFromForm]="langFormValue"
+  [displayResearchFromForm]="formResearch"
+  (newResearchEvent)="onNewResearch($event)">
+</app-display>
```

Comment:
Do you see how this is clean? ;) 
However, I am not sure the display should be part of the form